### PR TITLE
add pulsar output plugin

### PR DESCRIPTION
**Pull request by JesusIslam**
Add output plugin for [Apache Pulsar](https://pulsar.apache.org).

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by JesusIslam** [comment]
Turned out this needs the C++ lib. Will find a way to change to pure Go.

File: plugins/outputs/pulsar/pulsar.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		conf.InitialReconnectDelay, err = time.ParseDuration(p.InitialReconnectDelay)
+	}
+	if p.MaxReconnectDelay != "" {
+		conf.MaxReconnectDelay, err = time.ParseDuration(p.MaxReconnectDelay)
+	}
+	if p.NewProducerTimeout != "" {
+		conf.NewProducerTimeout, err = time.ParseDuration(p.NewProducerTimeout)
+	}
+	if p.SendTimeout != "" {
+		p.sendTimeout, err = time.ParseDuration(p.SendTimeout)
```
Comment (glinton) [inline plugins/outputs/pulsar/pulsar.go]: This was already done on line 78

File: plugins/outputs/pulsar/pulsar.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	} else {
+		p.sendTimeout = time.Second * 10
```
Comment (glinton) [inline plugins/outputs/pulsar/pulsar.go]: Set this default in `init()` and make sure the sample/example configs match

File: plugins/outputs/pulsar/pulsar.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+		buf, err := p.serializer.Serialize(metric)
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), p.sendTimeout)
+		defer cancel()
+		_, err = p.producer.Send(ctx, buf)
+		if err != nil {
+			return fmt.Errorf("FAILED to send Pulsar message: %s", err)
```
Comment (glinton) [inline plugins/outputs/pulsar/pulsar.go]: Don't return a capitalized error - https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by liangyuanpeng** [comment]
Hello guys,I am also working for outputs of pulsar, what is the current progress of this pr?

**Comment by sjwang90** [comment]
Closing this issue in favor of #7372 that uses pure Go pulsar client and doesn't require to installation of the C++ pulsar client

**Pull request closed without merging**